data_acquisition/py/gee_functions.py

```diff
@@ -323,6 +302,121 @@ def remove_geo(image):
   return image.setGeometry(None)
 
 
+def apply_fill_mask(image):
+  """ mask any fill values (0) in scaled raster
+  
+  Args:
+      image: ee.Image of an ee.ImageCollection
+
+  Returns:
+      an ee.Image where any values previously 0 are masked
+  """
+  b1_mask = image.select('SR_B1').gt(0)
+  b2_mask = image.select('SR_B2').gt(0)
+  b3_mask = image.select('SR_B3').gt(0)
+  b4_mask = image.select('SR_B4').gt(0)
+  b5_mask = image.select('SR_B5').gt(0)
+  b7_mask = image.select('SR_B7').gt(0)
+  fill_mask = (b1_mask.eq(1)
+    .And(b2_mask.eq(1))
+    .And(b3_mask.eq(1))
+    .And(b4_mask.eq(1))
+    .And(b5_mask.eq(1))
+    .And(b7_mask.eq(1))
+    .selfMask())
+  return image.updateMask(fill_mask.eq(1))
+
+
+# This should be applied AFTER scaling factors
+# Mask values less than -0.01
+def apply_realistic_mask(image):
+  """ mask out unrealistic SR values (those less than -0.01)
+  
+  Args:
+      image: ee.Image of an ee.ImageCollection
+
+  Returns:
+      an ee.Image where any re-scaled values <-0.01 are masked
+  """
+  b1_mask = image.select('SR_B1').gt(-0.01)
+  b2_mask = image.select('SR_B2').gt(-0.01)
+  b3_mask = image.select('SR_B3').gt(-0.01)
+  b4_mask = image.select('SR_B4').gt(-0.01)
+  b5_mask = image.select('SR_B5').gt(-0.01)
+  b7_mask = image.select('SR_B7').gt(-0.01)
+  realistic = (b1_mask.eq(1)
+    .And(b2_mask.eq(1))
+    .And(b3_mask.eq(1))
+    .And(b4_mask.eq(1))
+    .And(b5_mask.eq(1))
+    .And(b7_mask.eq(1))
+    .selfMask())
+  return image.updateMask(realistic.eq(1))
```

Comment:
suggestion: similar comment to the one prior, but it seems this is a function based on the 4/5/7 bands and it might be worth making a separate function for the 8/9 bands and just including SR_B6.

Reply:
Absolutely agree. Making changes now.